src/services/events/events-schema.ts

```diff
@@ -0,0 +1,36 @@
+import mongoose from "mongoose";
+import { Schema } from "mongoose";
+import { z } from "zod";
+
+export const EventInfo = z.object({
+    eventId: z.coerce.string().cuid2(),
+    name: z.string(),
+    date: z.string().transform((str) => new Date(str)),
+    start_time: z.string().transform((str) => new Date(str)),
+    end_time: z.string().transform((str) => new Date(str)),
```

Comment:
use the default zod datetime [object](https://zod.dev/?id=iso-date-time-amp-datetime-validation) - these can be 
```suggestion
    start_time: z.coerce().string().datetime(),
    end_time: z.coerce().string().datetime(),
```